include/oneapi/dpl/pstl/hetero/algorithm_ranges_impl_hetero.h

```diff
@@ -152,30 +152,47 @@ struct __swap1_wrapper;
 template <typename _Name>
 struct __swap2_wrapper;
 
-template <typename _BackendTag, typename _ExecutionPolicy, typename _Range1, typename _Range2, typename _Function>
+template <typename _BackendTag, typename _ExecutionPolicy, typename _Range1, typename _Range2>
 oneapi::dpl::__internal::__difference_t<_Range1>
-__pattern_swap(__hetero_tag<_BackendTag>, _ExecutionPolicy&& __exec, _Range1&& __rng1, _Range2&& __rng2, _Function __f)
+__pattern_swap(__hetero_tag<_BackendTag>, _ExecutionPolicy&& __exec, _Range1&& __rng1, _Range2&& __rng2)
 {
+    //a trivial pre-check
+    if (__rng1.empty() || __rng2.empty())
+        return 0;
```

Comment:
Please leave a comment why calling the method `empty()` is valid in this place.

Reply:
The usage of `std::ranges::empty()` looks more correct in the common case due it's implementation:
https://www.en.cppreference.com/w/cpp/ranges/empty.html
```
_A call to ranges::empty is expression-equivalent to:

bool(t.empty()), if that expression is valid.
Otherwise, (ranges::size(t) == 0), if that expression is valid.
Otherwise, bool(ranges::begin(t) == ranges::end(t)), if that expression is valid and decltype(ranges::begin(t)) models std::forward_iterator.
In all other cases, a call to ranges::empty is ill-formed, which can result in substitution failure when ranges::empty(t) appears in the immediate context of a template instantiation._
```
Fixed.